### DISCUSSION TITLE
How to call activity based on some conditions or some field value of previous activity output?

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5245] [Q&A]
Consider I have Workflow
public class CustomWorkFlow: IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
           .StartWith<Activity1>();
           .Then<Activity2>();
    }
}

Activity2 should get called based, if some condition satify which will be based on output of previous activity or based on some configurations present in config.json file
Suggest best approach to trigger activity conditionally

**Comment by glime-ai** [comment]
To trigger `Activity2` conditionally based on the output of `Activity1` or some configurations present in a `config.json` file, you can use the `If` activity or a similar conditional construct provided by Elsa. Here's a general approach to achieve this:

1. Define an `Outcome` from `Activity1` that represents the condition you want to check.
2. Use the `If` activity (or a custom conditional activity if `If` is not available) after `Activity1` to evaluate the condition.
3. In the `If` activity, you can access the output of `Activity1` and/or read the configurations from the `config.json` file to determine which branch to take.
4. Connect `Activity2` to the appropriate branch of the `If` activity.

Here's an example of how you might define this in your workflow:

```csharp
public class CustomWorkFlow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
                .Output(activity1Output => activity1Output) // Capture the output of Activity1
            .Then<IfElse>(ifElse =>
            {
                // Configure the IfElse activity to evaluate the condition
                ifElse
                    .When(context =>
                    {
                        // Access the output of Activity1
                        var output = context.Get(activity1Output);
                        // Read the configuration from the config.json file
                        var configValue = context.GetConfigValue("SomeConfigKey");
                        // Evaluate the condition based on the output and configuration
                        return output.SomeProperty == configValue;
                    })
                    .Then<Activity2>(); // If the condition is true, execute Activity2
            });
    }
}
```

In this example, `IfElse` is a hypothetical conditional activity that you would need to implement if Elsa does not provide one out of the box. The `context.GetConfigValue("SomeConfigKey")` is a placeholder for the actual method you would use to read from your `config.json` file.

Please note that the actual implementation may vary depending on the version of Elsa you are using and the specific requirements of your workflow. If Elsa provides an `If` activity or a similar construct, you should use that. Otherwise, you may need to create a custom activity to handle the conditional logic.

Remember to register your workflow and any custom activities with the Elsa services collection as needed.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by NiveditaGalwade** [reply]
.Output(activity1Output => activity1Output) // Capture the output of Activity1
 
 I am getting error here
 
 Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1061	'IActivityBuilder' does not contain a definition for 'Output' and no accessible extension method 'Output' accepting a first argument of type 'IActivityBuilder' could be found (are you missing a using directive or an assembly reference?)

**Reply by NiveditaGalwade** [reply]
@sfmskywalker can you please suggest me solution for this query?
